packages/dendron-plugin-views/src/hooks/useGraphElements.tsx

```diff
@@ -34,166 +35,333 @@ const DEFAULT_CLASSES = ""; //graph-view
-        fname: activeNote.fname,
-        color: getNoteColor({ fname: activeNote.fname, notes }),
-        stub: _.isUndefined(activeNote.stub) ? false : activeNote.stub,
-        localRoot: true,
-      },
-      classes: `${DEFAULT_NODE_CLASSES} ${getVaultClass(activeNote.vault)}`,
-      selected: true,
-    },
-  ];
+  // Breadth First Search for Descendants
+  while (nodesQueue.length > 0) {
+    const data = nodesQueue.dequeue();
 
-  // Add parent node
-  if (parentNote) {
+    if (!data) {
+      break;
+    }
+
+    const { note, distance, classes } = data;
+    const isActive = activeNote && note.id === activeNote.id;
     nodes.push({
       data: {
-        id: parentNote.id,
-        label: parentNote.title,
+        id: note.id,
+        label: note.title,
         group: "nodes",
-        fname: parentNote.fname,
-        color: getNoteColor({ fname: parentNote.fname, notes }),
-        stub: _.isUndefined(parentNote.stub) ? false : parentNote.stub,
+        fname: note.fname,
+        color: getNoteColor({ fname: note.fname, notes }),
+        stub: _.isUndefined(note.stub) ? false : note.stub,
+        localRoot: distance === 0,
       },
-      classes: `${DEFAULT_NODE_CLASSES} parent ${getVaultClass(
-        parentNote.vault
-      )}`,
+      classes,
+      selected: isActive,
     });
-  }
 
-  // Add connected nodes
-  nodes.push(
-    ...Object.values(connectedNotes).map((note) => {
-      return {
-        data: {
-          id: note.id,
-          label: note.title,
-          group: "nodes",
-          fname: note.fname,
-          color: getNoteColor({ fname: note.fname, notes }),
-          stub: _.isUndefined(note.stub) ? false : note.stub,
-        },
-        classes: `${DEFAULT_NODE_CLASSES} ${getVaultClass(note.vault)}`,
-      };
-    })
-  );
+    if (data.distance < maxDistance) {
+      const noteVaultClass = getVaultClass(note.vault);
+
+      //Add parents and grandparents. Distance 0 indicates the immediate parent, isParent indicate the ancestor
+      const parentNote =
+        (data.distance === 0 || data.isParent) && note.parent
+          ? notes[note.parent]
+          : undefined;
+      if (parentNote) {
+        nodesQueue.enqueue({
+          note: parentNote,
+          distance: data.distance + 1,
+          isParent: true,
+          classes: `${DEFAULT_NODE_CLASSES} parent ${getVaultClass(
+            parentNote.vault
+          )}`,
+        });
+        edges.hierarchy.push({
+          data: {
+            group: "edges",
+            id: `${parentNote.id}_${note.id}`,
+            source: parentNote.id,
+            target: note.id,
+            fname: parentNote.fname,
+            stub: _.isUndefined(parentNote.stub) ? false : parentNote.stub,
+          },
+          classes: `${DEFAULT_EDGE_CLASSES} hierarchy ${noteVaultClass}`,
+        });
+      }
+      // do not include children of parent/grandparent
+      const children = data.isParent ? [] : note.children;
```

Comment:
nit: calling this children is a bit misleading.  Maybe better would be to do something like 

```typescript
// do not include children of parent/grandparent
if(!data.isParent) {
children.forEach(...);
}
```